### PR TITLE
Don't replace ampersands in X-PJAX-URLs

### PR DESCRIPTION
**Pull request by tdhooper**
Urls containing ampersands, for example http://www.foo.com/some-&-path were getting their ampersand replaced with a ?.

I've put my reasoning for the change in the commit message, I'd like make sure there's no behaviour that I'm breaking here.

### CODE REVIEW

**Comment by mislav** [comment]
Yikes, good catch. I didn't anticipate that pathnames might have ampersands in them.

File: jquery.pjax.js [hunk 1]
```diff
@@ -576,7 +576,6 @@ function stripParam(url, name) {
   return url
     .replace(new RegExp('[?&]' + name + '=[^&#]*'), '')
     .replace(/[?&]($|#)/, '\1')
-    .replace(/[?&]/, '?')
```
Comment (mislav) [inline jquery.pjax.js]: This line was necessary. I don't know why removing it didn't break any tests. It's here for this case:

1. URL is `foo?_pjax=xy&bar=baz`
2. `_pjax` gets stripped: `foo&bar=baz`
3. Then I transform the first `?` or `&` character into `?`, to handle the string that was result of step (2): `foo?bar=baz`

File: test/unit/pjax.js [hunk 2]
```diff
@@ -32,6 +32,22 @@ if ($.support.pjax) {
     })
   })
 
+  asyncTest("copes with ampersands when pushing urls", function() {
+    var frame = this.frame
+
+    frame.$('#main').on('pjax:success', function() {
+      console.log(frame.location.pathname)
```
Comment (mislav) [inline test/unit/pjax.js]: `console.log` sneaked in

File: test/unit/pjax.js [hunk 3]
```diff
@@ -32,6 +32,22 @@ if ($.support.pjax) {
...
+      equal(frame.location.pathname, "/some-&-path/hello.html")
+      equal(frame.location.search, "")
+      start()
+    })
+    frame.$.pjax({
+      url: "/some-&-path/hello.html",
+      container: "#main",
+      cache: false
```
Comment (mislav) [inline test/unit/pjax.js]: `cache: false` is not necessary here

File: test/unit/pjax.js [hunk 4]
```diff
@@ -32,6 +32,22 @@ if ($.support.pjax) {
     })
   })
 
+  asyncTest("copes with ampersands when pushing urls", function() {
```
Comment (mislav) [inline test/unit/pjax.js]: Maybe rename the test to something like "compatible with pathnames that contain ampersand"?

File: jquery.pjax.js (hunk 1 above)
Comment (tdhooper) [inline jquery.pjax.js]: In what cases do the _ or _pjax parameters end up first in query string? I assumed they would always be appended.

Comment (mislav) [inline jquery.pjax.js]: I didn't give that a thought before, but you're right that the `_pjax` and `_` params will [likely always be appended](https://github.com/defunkt/jquery-pjax/blob/c4945b67869874e52cd19f4e1c900e3da5fdd2f4/jquery.pjax.js#L188-193). I just wanted my param stripping code to be agnostic of this fact.

We could definitely improve this function so it handles all cases well

**Pull request closed without merging**